src/main/java/Ally/Parser.java

```diff
@@ -0,0 +1,43 @@
+package Ally;
+
+import Ally.Commands.*;
+import Ally.Exceptions.AllyException;
+
+public class Parser {
+
+    /**
+     * Returns the different Commands according to the user input.
+     * If there is no input matching the conditions, AllyException will be thrown.
+     *
+     * @param ipt
+     * @return
+     * @throws AllyException
+     */
+    public static Commands parse(String ipt) throws AllyException {
```

Comment:
```suggestion
    public static Commands parse(String input) throws AllyException {
```
Perhaps you can use standard words to improve clarity.